src/Data/V/Linear/Internal/V.hs

```diff
@@ -61,115 +62,74 @@ structure.
 -------------------------------------------------------------------------------
 
 newtype V (n :: Nat) (a :: Type) = V (Vector a)
-  deriving (Eq, Ord, Prelude.Functor)
+  deriving (Prelude.Eq, Prelude.Ord, Prelude.Functor)
 
 -- Using vector rather than, say, 'Array' (or directly 'Array#') because it
 -- offers many convenience function. Since all these unsafeCoerces probably
 -- kill the fusion rules, it may be worth it going lower level since I
 -- probably have to write my own fusion anyway. Therefore, starting from
 -- Vectors at the moment.
 
-type family FunN (n :: Nat) (a :: Type) (b :: Type) :: Type where
-  FunN 0 a b = b
-  FunN n a b = a %1 -> FunN (n - 1) a b
-
 -- # API
 -------------------------------------------------------------------------------
 
-theLength :: forall n. KnownNat n => Int
-theLength = fromIntegral (natVal' @n (proxy# @_))
+consume :: V 0 a %1 -> ()
+consume = Unsafe.toLinear (\_ -> ())
+
+-- Backward compatility
+-- consumeV :: V 0 a %1 -> b %1 -> b
+-- consumeV v b = consume v & \case
+--   () -> b
+
```

Comment:
Remove the dead code.
```suggestion
```

Reply:
So, we don't need to preserve the old API, right?

Reply:
Right.